src/components/auth/signIn/email/SignInModalContent.tsx

```diff
@@ -0,0 +1,290 @@
+import HCaptcha from '@hcaptcha/react-hcaptcha'
+import { isStr } from '@subsocial/utils'
+import { Button, Form, FormInstance, Input } from 'antd'
+import clsx from 'clsx'
+import jwtDecode from 'jwt-decode'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { MutedDiv } from 'src/components/utils/MutedText'
+import {
+  emailSignIn,
+  JwtPayload,
+  onErrorHandler,
+} from 'src/components/utils/OffchainSigner/api/requests'
+import { setAuthOnRequest } from 'src/components/utils/OffchainSigner/api/utils'
+import { hCaptchaSiteKey } from 'src/config/env'
+import { VALIDATION_DELAY } from 'src/config/ValidationsConfig'
+import useSignerExternalStorage from 'src/hooks/useSignerExternalStorage'
+import { debounceFunction } from 'src/utils/async'
+import { StepsEnum, useAuth } from '../../AuthContext'
+import styles from './SignInModalContent.module.sass'
+import { RegexValidations, useFormValidation } from './useFormValidation'
+
+type FormValues = {
+  email: string
+  password: string
+}
+
+type FieldName = keyof FormValues
+
+export const fieldName = (name: FieldName): FieldName => name
+
+type Props = {
+  setCurrentStep: (step: number) => void
+  onSignInSuccess: (address: string) => void
+}
+
+type InputProps = {
+  error: string | null
+  isError: boolean
+  form: FormInstance<FormValues>
+  data?: string
+}
+
+export const EmailInput = ({ data, error, isError, form }: InputProps) => {
+  const validateEmail = () => {
+    form.validateFields(['email'])
+  }
+  const debouncedValidateField = useCallback(debounceFunction(validateEmail, VALIDATION_DELAY), [])
+
+  const handleAfterInput = (value: string) => {
+    form.setFieldsValue({ [fieldName('email')]: value.trim() })
+    debouncedValidateField()
+  }
+
+  return (
+    <Form.Item
+      initialValue={data}
+      name={fieldName('email')}
+      className={clsx(styles.BaseFormItem, isError && styles.BaseFormItemError)}
+      rules={[
+        {
+          pattern: RegexValidations.ValidEmail,
+          message: 'Please enter a valid email address.',
+          validateTrigger: ['onBlur'],
+        },
+      ]}
+      validateStatus={isStr(error) ? 'error' : undefined}
+    >
+      <Input
+        autoFocus
+        required
+        type='email'
+        placeholder='Email'
+        onChange={e => {
+          handleAfterInput(e.target.value)
+        }}
+        onBlur={e => {
+          handleAfterInput(e.target.value)
+        }}
```

Comment:
Same code, extract to function

Reply:
Extracted in [f667afb](https://github.com/dappforce/polkaverse/pull/74/commits/f667afb94d51c32e4334f6d041437ee2c4d2e829)